### PR TITLE
Add sbom section to the security insights file

### LINKED ISSUE #1470: Add dependencies/sbom section to SECURITY-INSIGHTS.yml
`#SecuritySlam`

**What would you like to be cleaned**:

When the SECURITY-INSIGHTS.yml file was initially created in #1469, the dependencies/sbom section was omitted, as the project was not creating sboms at that time. Once sboms are being created, the SECURITY-INSIGHTS.yml file should be updated to include this section.

**Why is this needed**:

Adding this section will improve the project's score on the CLOMonitor site.

### PR DESCRIPTION
**Pull request by mwysokin**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #1470 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mwysokin** [comment]
I have no idea whether I correctly identified the scope as stated here: https://github.com/kubernetes-sigs/kueue/issues/1470#issuecomment-2178377246 Please let me know if this is something expected and what might be missing 🙏

**Comment by alculquicondor** [comment]
/ok-to-test

LGTM, please just add a note in `.github/ISSUE_TEMPLATE/NEW_RELEASE.md` to update this field during a release.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ Please do not remove items from the checklist
       Link: <!-- example https://github.com/kubernetes-sigs/kueue/releases/tag/v0.1.0 -->
 - [ ] Run the [openvex action](https://github.com/kubernetes-sigs/kueue/actions/workflows/openvex.yaml) to generate openvex data. The action will add the file to the release artifacts.
 - [ ] Run the [SBOM action](https://github.com/kubernetes-sigs/kueue/actions/workflows/sbom.yaml) to generate the SBOM and add it to the release.
+- [ ] Update the value of the key `dependencies.sbom.sbom-file` in `SECURITY-INSIGHTS.yaml` with the SBOM generated in the previous step.
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:50]: line 58 already lists other changes to the same file. Let's put it there.

Comment (mwysokin) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: Good point. I didn't realize the file is already covered elsewhere 😅

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**